django/core/management/commands/dbshell.py

```diff
@@ -15,11 +15,13 @@ def add_arguments(self, parser):
             '--database', default=DEFAULT_DB_ALIAS,
             help='Nominates a database onto which to open a shell. Defaults to the "default" database.',
         )
+        parameters = parser.add_argument_group('parameters', prefix_chars='--')
+        parameters.add_argument('parameters', nargs='*')
 
     def handle(self, **options):
         connection = connections[options['database']]
         try:
-            connection.client.runshell()
+            connection.client.runshell(options['parameters'])
```

Comment:
I wonder if we should suppress `subprocess.CalledProcessError` when arguments are not correct, to get e.g.
```
/usr/lib/postgresql/10/bin/psql: nieznana opcja '--commandasdasd'
Try "psql --help" for more information.
```
instead of
```
/usr/lib/postgresql/10/bin/psql: nieznana opcja '--commandasdasd'
Try "psql --help" for more information.
Traceback (most recent call last):
  File "manage.py", line 22, in <module>
    main()
  File "manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "django/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "django/django/core/management/base.py", line 328, in run_from_argv
    self.execute(*args, **cmd_options)
  File "django/django/core/management/base.py", line 369, in execute
    output = self.handle(*args, **options)
  File "django/django/core/management/commands/dbshell.py", line 25, in handle
    connection.client.runshell(options['parameters'])
  File "django/django/db/backends/postgresql/client.py", line 55, in runshell
    self.runshell_db(self.connection.get_connection_params(), parameters)
  File "django/django/db/backends/postgresql/client.py", line 49, in runshell_db
    subprocess.run(args, check=True, env=subprocess_env)
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['psql', '-U', 'djangoticket', '-h', 'localhost', '-p', '5432', 'djangoticket', '--commandasdasd']' returned non-zero exit status 1.
```

We can also re-raise it as a `CommandError`:
```
/usr/lib/postgresql/10/bin/psql: nieznana opcja '--commandasdasd'
Try "psql --help" for more information.
CommandError: Command '['psql', '-U', 'djangoticket', '-h', 'localhost', '-p', '5432', 'djangoticket', '--commandasdasd']' returned non-zero exit status 1.
```

Reply:
I'm in favor of re-raising `subprocess.CalledProcessError` as a `CommandError`, e.g.
```python
    def handle(self, **options):
        connection = connections[options['database']]
        try:
            connection.client.runshell(options['parameters'])
        except OSError:
            # Note that we're assuming OSError means that the client program
            # isn't installed. There's a possibility OSError would be raised
            # for some other reason, in which case this error message would be
            # inaccurate. Still, this message catches the common case.
            raise CommandError(
                'You appear not to have the %r program installed or on your path.' %
                connection.client.executable_name
            )
        except subprocess.CalledProcessError as e:
            raise CommandError(
                '"%s" returned non-zero exit status %s.' % (
                    ' '.join(e.cmd),
                    e.returncode),
            )
```
which will end with:
```
/usr/lib/postgresql/10/bin/psql: nieznana opcja '--commandasdasd'
Try "psql --help" for more information.
CommandError: "psql -U djangoticket -h localhost -p 5432 djangoticket --asdasdad" returned non-zero exit status 1.
```